src/common/ChannelChatters.cpp

```diff
@@ -64,9 +64,31 @@ void ChannelChatters::addPartedUser(const QString &user)
         });
     }
 }
+
+const QColor ChannelChatters::getUserColor(const QString &user)
+{
+    auto chatterColors = this->chatterColors_.access();
+
+    auto search = chatterColors->find(user.toLower());
+    if (search == chatterColors->end())
+    {
+        return QColor(MessageColor::Text);
+    }
+    else
+    {
+        return search->second;
+    }
+}
+
 void ChannelChatters::setChatters(UsernameSet &&set)
 {
     *this->chatters_.access() = set;
 }
 
+void ChannelChatters::setUserColor(const QString &user, const QColor color)
+{
+    auto chatterColors = this->chatterColors_.access();
+    chatterColors->emplace(user.toLower(), color);
```

Comment:
When a user changes their user name color, this change will not be respected by `chatterColors`.
[`std::map::emplace` only inserts the newly constructed element if there is no element with the key in the container.](https://en.cppreference.com/w/cpp/container/map/emplace)

(It's a niche scenario but I think we should still put thought into it.)